### DISCUSSION TITLE
SUGGESTION: a new once() helper function

### DISCUSSION THREAD

**Discussion by basepack** [spatie/ray#473] [Ideas]
Imagine this loop:

```php 
$orders = Order::take(100)->get();

$orders->each(function(Order $order){
   $complexStuff = new ComplexAction()->execute($order);

   ray($complexStuff);
});
```

That will send it to ray a 100 times and ray will have difficulties displaying this much information. 

However if we could have a `once()` helper which could be used like this: `ray($complexStuff)->once();` we can easily prevent the other 99 times we would not need there.

One could argue that I should use `ray()->die()` for this case, but I will not because my BrowserSync is then not working anymore and also when debugging Laravel Livewire I run into trouble.

**Comment by freekmurze** [comment]
I like that! 

@patinthehat Would you like to implement this? If you got no time, I'll take care of this myself soon.

**Reply by basepack** [reply]
Thank you Freek! 

Astonished by your response speed every time, keep it up! :)

**Reply by patinthehat** [reply]
> @patinthehat Would you like to implement this? If you got no time, I'll take care of this myself soon.

@freekmurze Sure! I'll implement this.

**Reply by patinthehat** [reply]
@basepack This feature has been added as of package version `1.27.0`.

**Reply by basepack** [reply]
@patinthehat Many thanks!
